py-polars/polars/lazyframe/frame.py

```diff
@@ -2798,7 +2798,7 @@ def filter(
             elif p is True:
                 continue  # no-op; matches all rows
             if _is_generator(p):
-                p: Iterable[IntoExprColumn] = tuple(p)
+                p = tuple(p)  # type: ignore[arg-type]
```

Comment:
Not super happy with this, but I think I'd need to make some changes in `_utils.various` to keep the typing. 

Can add that in another commit if needed

Reply:
It would be pretty easy if you just change the return type to ` TypeGuard[Generator[Any, Any, Any]]:`